java/system-test/system-test-suite/src/main/java/sleeper/systemtest/suite/dsl/SleeperSystemTest.java

```diff
@@ -38,28 +51,58 @@ public class SleeperSystemTest {
     private final SystemTestParameters parameters = SystemTestParameters.loadFromSystemProperties();
     private final CloudFormationClient cloudFormationClient = CloudFormationClient.create();
     private final AmazonS3 s3Client = AmazonS3ClientBuilder.defaultClient();
-    private final AmazonDynamoDB dynamoDB = AmazonDynamoDBClientBuilder.defaultClient();
+    private final S3Client s3ClientV2 = S3Client.create();
+    private final AmazonDynamoDB dynamoDBClient = AmazonDynamoDBClientBuilder.defaultClient();
+    private final AmazonSQS sqsClient = AmazonSQSClientBuilder.defaultClient();
+    private final LambdaClient lambdaClient = createSystemTestLambdaClient();
     private final SleeperInstanceContext instance = new SleeperInstanceContext(
-            parameters, cloudFormationClient, s3Client, dynamoDB);
+            parameters, cloudFormationClient, s3Client, dynamoDBClient);
+    private final IngestSourceFilesContext sourceFiles = new IngestSourceFilesContext(parameters, s3ClientV2);
+
+    public SleeperSystemTest() {
+        sourceFiles.createOrEmptySourceBucket();
+    }
 
     public static SleeperSystemTest getInstance() {
         return INSTANCE;
     }
 
     public void connectToInstance(SystemTestInstance testInstance) {
-        instance.connectTo(testInstance.getIdentifier(), testInstance.getInstanceConfiguration());
+        instance.connectTo(testInstance.getIdentifier(), testInstance.getInstanceConfiguration(parameters));
         instance.reinitialise();
     }
 
     public InstanceProperties instanceProperties() {
         return instance.getInstanceProperties();
     }
 
+    public void updateTableProperties(Consumer<TableProperties> tablePropertiesConsumer) {
+        tablePropertiesConsumer.accept(instance.getTableProperties());
+        try {
+            instance.getTableProperties().saveToS3(s3Client);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
     public SystemTestDirectIngest directIngest(Path tempDir) {
         return new SystemTestDirectIngest(new DirectIngestDriver(instance, tempDir));
     }
 
     public SystemTestDirectQuery directQuery() {
         return new SystemTestDirectQuery(new DirectQueryDriver(instance));
     }
+
+    public SystemTestSourceFiles sourceFiles() {
+        return new SystemTestSourceFiles(instance, sourceFiles);
+    }
+
+    public SystemTestIngestBatcher ingestBatcher() {
+        return new SystemTestIngestBatcher(this, parameters, instance,
+                new IngestBatcherDriver(instance, dynamoDBClient, sqsClient, lambdaClient));
+    }
```

Comment:
We could gather the ingest features into one class to limit the number of accessors we have here.